### PR TITLE
Fixes #15403: inlining not propagated along aliases of module paths in delta-resolver

### LINKED ISSUE #15403: Inlining in non-logical objects not granted in module definitions
#### Description of the problem

```coq
Module Type EqType.
   Parameter Inline t : Type.
   Parameter eq : t -> t -> Prop.
   Notation "#" := t (only parsing).
End EqType.

Module NAT.
   Definition t := nat.
   Definition eq := @eq nat.
End NAT.

Module F (E:EqType) :=
  E.

Module Import NAT' := F NAT.
About NAT'.eq. (* no arguments scopes, because it is for the non-logical-objects purpose of type NAT'.t *)
About NAT.eq. (* arguments scopes have also been canceled *)
Check #. (* no inlining: NAT'.t *)
```
Contrastingly, it is ok with an `Include`:
```coq
Module F (E:EqType).
  Include E.
End F.

Module Import NAT' := F NAT.
About NAT'.eq. (* Arguments NAT'.eq (_ _)%nat_scope *)
About NAT.eq. (* Arguments NAT.eq (_ _)%nat_scope *)
Check #. (* nat *)
```
(explicit qualification with `NAT'` even when imported is from me, to emphasize the differences)

Fix possibly to come.

#### Coq Version

From 8.5 to master.

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** fix

The invariants of the code for delta-resolver composition are complex and I don't understand them well. At least, there was a discrepancy when composing either a delta-resolver `[Top.NAT'=>Top.E]` or delta-resolver `[Top.NAT'.t=>Top.E.t]` with a substitution `{Top.E |-> Top.NAT [Top.NAT.t=>inline(Some _)]}`: the second case was adding `[Top.NAT'.t=>inline(Some _)]` but not the first case. The discrepancy was in `Mod_subst.gen_subst_delta_resolver`, which `subst_codom_delta_resolver` was calling and we fix the first case.

Fixes / closes #15403

- [x] Added / updated **test-suite**.
- [x] Added **changelog**.

### CODE REVIEW

**Review by SkySkimmer — APPROVED** [review]
@coq/kernel-maintainers can I get a second opinion?

**Comment by ppedrot** [comment]
This inlining feature is utter dark magic to me, and I have no idea of the expected invariants. I just happen to know you can bypass a lot of kernel checks thanks to it, so I'd advise caution.

**Comment by herbelin** [comment]
After two weeks looking at the code, I still don't get all the invariants either. They are quite loose, e.g. are the Δ-equations closed by transitivity, or arbitrarily mixing atomic steps (`kn` unfolds in one step to `kn'`) with normalized equations (`kn` Δ-nomalizes to `kn'`), with sparadrap here and there (e.g. the `Change_equiv_to_inline` exception), with questionable overriding of Δ-equations by inlining. A model based on an `Alias` field in constant/inductive bodies would look much simpler (would the extra indirections be a source of inefficiency? maybe not more than maintaining a big `mod_delta`?). But even in a model with an `Alias` field, it seems that inlining should first be disentagled from `mod_delta`.

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**
